.github/workflows/release.yml

```diff
@@ -13,10 +13,22 @@ jobs:
   release-please:
     runs-on: ubuntu-latest
     steps:
-      - uses: google-github-actions/release-please-action@v3
+      - id: release
+        uses: google-github-actions/release-please-action@v3
         with:
           release-type: node
           package-name: mongodb
-          pull-request-title-pattern: "chore${scope}: release${component} ${version} [skip-ci]"
-    # Publication steps would be dependent on the output of the above step:
-    # steps.release.outputs.release_created
+          pull-request-title-pattern: 'chore${scope}: release${component} ${version} [skip-ci]'
+          changelog-path: HISTORY.md
+          default-branch: main
+
+      # If release-please created a release, publish to npm
+      - if: ${{ steps.release.outputs.release_created }}
+        uses: actions/checkout@v3
+      - if: ${{ steps.release.outputs.release_created }}
+        name: actions/setup
+        uses: ./.github/actions/setup
+      - if: ${{ steps.release.outputs.release_created }}
+        run: npm publish --provenance
```

Comment:
- upon merging any commit to main: https://github.com/nbbeeken/node-mongodb-native/actions/runs/5281781520/jobs/9555817567
- running the history workflow targetting the release PR: xxx
- upon merging the release PR: xxx

Reply:
So the steps as this PR stands:
- every releasable unit (fix/feat) commit to main will cause release-please to create/update the release PR
  - It will overwrite the contents of the PR body indiscriminately
- We currently have to manually run the history action, it gathers the PRs involved in the release and posts the highlights defined in each one to the release PR
  - There's no strong reason to run this any earlier than when we're ready for releasing
  - We can run it early to preview how the highlights look when all gathered together
  - Essentially we use the PRs going into the release as storage for the text that is overwritten by release-please
- Merging the release PR causes the release, the body of the PR *between* the horizontal lines: (markdown `---`) is used for the body of our release.
  - I think we should have authors go back and edit their PRs & rerun the history action.
  - However, we can make last minute edits to the PR directly and it will be transferred to the GH release.

Reply:
> We currently have to manually run the history action, it gathers the PRs involved in the release and posts the highlights defined in each one to the release PR

Why can't we just regenerate the history as a part of generating/updating the release PR?

Reply:
> Merging the release PR causes the release, the body of the PR between the horizontal lines: (markdown ---) is used for the body of our release.
> - I think we should have authors go back and edit their PRs & rerun the history action.
> - However, we can make last minute edits to the PR directly and it will be transferred to the GH release.

When do PR authors edit their PRs?  Is that after we run the release action?

I just want this process to be very clear before we try it on a release with @nbbeeken OOO.

-----

It sounds like there are more steps than just
- PR author includes highlights in PR description, if necessary
- PR merges into main, triggering the action that regenerates history, pulls in the new highlights if necessary, and then updates the release notes
- Release PR is merged for release

What technical limitations are there that prevent the above workflow?

Reply:
also - how do docs factor into the release workflow?  Our current practice is to regenerate them and include them in the release commit.  Will they be auto-generated as a part of the release and how do we specify that we need to regenerate for a `minor` or major?

Reply:
> Why can't we just regenerate the history as a part of generating/updating the release PR?

I don't have direct access to the PR number & git ref that the auto release is pending at, finding it isn't exactly clear, we'd have to fetch all the PRs and filter, and I'm not sure on what.